File: python_benchmarks/Example1.py
'''
REQUIRED: have Cassandra DB already running
pip install cassandra-driver (https://docs.datastax.com/en/developer/python-driver/3.25/)

(This is run from the parent directory, so
"python_benchmarks.AbstractBenchmark"
is required instead of "AbstractBenchmark")
'''
from python_benchmarks.AbstractBenchmark import AbstractBenchmark
from cassandra.cluster import Cluster


class Example1(AbstractBenchmark):

    def __init__(self):
        self.category = 'Read'
        self.description = 'This is an example'

    def setupQuery(self):
        print('Connecting to cassandra...')
        cluster = Cluster()
        self.session = cluster.connect()

    def endQuery(self):
        print('Closing...')
        self.session.close()

    def runQuery(self):
        result = self.session.execute(
            'SELECT release_version FROM system.local')
        print(result.one())

File: java_benchmarks/src/main/java/com/cmpe220/benchmark/Example1.java
package com.cmpe220.benchmark;

import com.datastax.oss.driver.api.core.CqlSession;
import com.datastax.oss.driver.api.core.cql.*;

public class Example1 extends AbstractBenchmark {

    private CqlSession session;

    public Example1() {
        category = "Read";
        description = "This is an example";
    }

    public void setupQuery() {
        System.out.println("Connecting to cassandra...");
        session = CqlSession.builder().build();
    }

    public void endQuery() {
        System.out.println("Closing...");
        session.close();
    }

    public void runQuery() {
        ResultSet rs = session.execute("select release_version from system.local");
        Row row = rs.one();
        System.out.println(row.getString("release_version"));
    }
}
